lib/web_ui/lib/src/ui/painting.dart

```diff
@@ -508,19 +550,30 @@ void decodeImageFromPixels(
   int? targetHeight,
   bool allowUpscaling = true,
 }) {
-  final Future<Codec> codecFuture = _futurize((engine.Callback<Codec> callback) {
-    return _instantiateImageCodec(
+  if (engine.experimentalUseSkia) {
+    engine.skiaInstantiateImageCodec(
       pixels,
-      callback,
-      width: width,
-      height: height,
-      format: format,
-      rowBytes: rowBytes,
+      (Codec codec) {
+        codec.getNextFrame().then((FrameInfo info) {
+          callback(info.image);
+        });
+      },
+      width,
+      height,
+      format.index,
+      rowBytes,
     );
-  });
-  codecFuture
-      .then((Codec codec) => codec.getNextFrame())
-      .then((FrameInfo frameInfo) => callback(frameInfo.image));
+    return;
+  }
+  switch (format) {
+    case PixelFormat.rgba8888:
+      _createBmp(pixels, width, height, Endian.big, callback);
+      return;
+    case PixelFormat.bgra8888:
+      _createBmp(pixels, width, height, Endian.little, callback);
```

Comment:
I don't know enough about this but wouldn't the little-endian version of of rgba8888 be abgr8888? (i.e. _not_ bgra8888) Or is it just a naming issue?

Reply:
Ahhh, I think I need this to be working as if we're on Uint16 with the endianness swapped .. in which case, I might as well just do the swapping myself I think.

Reply:
I also forgot to handle rowbytes.